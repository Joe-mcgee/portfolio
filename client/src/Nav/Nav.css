.Nav {
  width: 100%;
  height: 100%;
  background: radial-gradient(closest-side at 62% 38%, rgba(255,255,255,0.62), rgba(255,255,255, 0.38));
}

.Nav:hover {
  background: radial-gradient(closest-side at 62% 38%, rgba(218, 165, 32, 0.62), rgba(218, 165, 32, 0.38));
}