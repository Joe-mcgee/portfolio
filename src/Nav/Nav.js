import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {BrowserRouter, NavLink } from 'react-router-dom';

class Nav extends Component {
  constructor(props) {
    super(props)
    this.state = {

    }
  }

  render() {
    return (
      <nav className='Nav'>
        <ul className='Nav__container'>
          <li><NavLink to='/' exact className='Nav__home'>Home</NavLink></li>
          <li><NavLink to='/blog' className='Nav__blog'>Blog</NavLink></li>
          <li><NavLink to='/portfolio' className='Nav__Portfolio'>Portfolio</NavLink></li>
        </ul>
      </nav>
      )
  };
}


Nav.propTypes = {

}

export default Nav