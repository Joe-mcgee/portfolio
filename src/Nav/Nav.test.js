import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import './Nav.css';
import {BrowserRouter, NavLink} from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
    <Nav />
    </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
})

describe('NavLinks', () => {

  it('should have a home route', () => {
    const home = renderer.create(<BrowserRouter><NavLink to='/' exact className='Nav__home'>Home</NavLink></BrowserRouter>)
    expect(home).toMatchSnapshot()
  })

  it('should have a blog route', () => {
    const blog = renderer.create(<BrowserRouter><NavLink to='/blog' exact className='Nav__blog'>Blog</NavLink></BrowserRouter>)
  })

})