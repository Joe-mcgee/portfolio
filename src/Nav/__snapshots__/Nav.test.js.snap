// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks should have a blog route 1`] = `
<a
  aria-current={false}
  className="Nav__blog"
  href="/blog"
  onClick={[Function]}
  style={undefined}
>
  Blog
</a>
`;

exports[`NavLinks should have a home route 1`] = `
<a
  aria-current="true"
  className="Nav__home active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  Home
</a>
`;
